### PR TITLE
Fix uninstalling modules with deleted hooks.

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | If a hook was removed from database, you can no longer uninstall a module or unhook it from there. The only change is basically adding a try catch, renaming a variable for better clarity and some comments.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See below
| Fixed issue or discussion?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/29483
| Related PRs       | 
| Sponsor company   |

### How to reproduce
1. Install [gsitemap](https://github.com/PrestaShop/gsitemap) native module. It adds a new hook `gSitemapAppendUrls`.
2. Install this [tox_hooktest.zip](https://github.com/PrestaShop/PrestaShop/files/12498038/tox_hooktest.zip), it hooks itself to `gSitemapAppendUrls`.
3. Uninstall gsitemap - it removes `gSitemapAppendUrls`.
4. Try to uninstall the test module - `Cannot Uninstall module tox_hooktest. The hook id #405 does not exist in database.`

### CODE REVIEW

File: classes/Hook.php [hunk 1]
```diff
@@ -637,21 +637,52 @@ public static function registerHook($module_instance, $hook_name, $shop_list = n
...
+             * It would be better if this method returned false in future versions.
+             */
+            try {
+                $hook_name = Hook::getNameById((int) $hook_identifier);
+            } catch (Exception $e) {
+            }
+
+            if (empty($hook_name)) {
+                $hook_name = '';
+            }
```
Comment (jolelievre) [inline classes/Hook.php]:
```suggestion
            } catch (PrestaShopObjectNotFoundException $e) {
                $hook_name = '';
            }
```

File: classes/Hook.php [hunk 2]
```diff
@@ -637,21 +637,52 @@ public static function registerHook($module_instance, $hook_name, $shop_list = n
...
         }
 
-        Hook::exec('actionModuleUnRegisterHookBefore', ['object' => $module_instance, 'hook_name' => $hook_name]);
+        Hook::exec(
+            'actionModuleUnRegisterHookBefore',
+            [
+                'object' => $module_instance,
+                'hook_name' => $hook_name,
+            ]
+        );
```
Comment (jolelievre) [inline classes/Hook.php]:
```suggestion
        if (!empty($hook_name)) {
            Hook::exec(
                'actionModuleUnRegisterHookBefore',
                [
                    'object' => $module_instance,
                    'hook_name' => $hook_name,
                ]
            );
        }
```

File: classes/Hook.php (hunk 1 above)
Comment (Hlavtox) [reply classes/Hook.php]: Done, but I didn't move the `$hook_name = '';`, because we could get an empty string.

File: classes/Hook.php (hunk 2 above)
Comment (Hlavtox) [reply classes/Hook.php]: Done, also for the second call. :-)

**Review by kpodemski — DISMISSED** [review]

File: classes/Hook.php [hunk 3]
```diff
@@ -637,21 +637,55 @@ public static function registerHook($module_instance, $hook_name, $shop_list = n
...
+     *
+     * @return bool
+     */
+    public static function unregisterHook($module_instance, $hook_identifier, $shop_list = null)
     {
-        if (is_numeric($hook_name)) {
-            // $hook_name passed it the id_hook
-            $hook_id = $hook_name;
-            $hook_name = Hook::getNameById((int) $hook_id);
+        if (is_numeric($hook_identifier)) {
```
Comment (PululuK) [inline classes/Hook.php]:
```suggestion
        if (Validate::isUnsignedInt($hook_identifier)) {
```

**Review by LouisAUTHIE — APPROVED** [review]

**Review by boherm — APPROVED** [review]
LGTM, thanks @Hlavtox!

**Review by jolelievre — APPROVED** [review]
Thanks @Hlavtox

**Review by MhiriFaten — APPROVED** [review]
Hello @Hlavtox ,

I have checked your PR and the uninstalling modules with deleted hooks problem is fixed.
Automated tests are OK.

**Comment by nicosomb** [comment]
Thank you @Hlavtox

**Pull request merged**